currency/manager_test.go

```diff
@@ -574,6 +575,41 @@ func TestFullStoreUnmarshalMarshal(t *testing.T) {
 	}
 }
 
+func TestIsPairAvailable(t *testing.T) {
+	t.Parallel()
+	pm := initTest(t)
+	cp := NewPairWithDelimiter("BTC", "USD", "-")
+	ok, err := pm.IsPairAvailable(cp, asset.Spot)
+	require.NoError(t, err, "IsPairAvailable must not error")
+	assert.True(t, ok, "IsPairAvailable should return correct value for an available and enabled pair")
+
+	ok, err = pm.IsPairEnabled(NewPair(SAFE, MOONRISE), asset.Spot)
+	require.NoError(t, err, "IsPairAvailable must not error")
+	assert.False(t, ok, "IsPairAvailable should return correct value for an non-existent")
+
+	ok, err = pm.IsPairEnabled(cp, asset.Futures)
+	require.NoError(t, err, "IsPairAvailable must not error")
+	assert.False(t, ok, "IsPairAvailable should return false for a disabled asset type")
+
+	cp = NewPairWithDelimiter("XRP", "DOGE", "-")
+	ok, err = pm.IsPairEnabled(cp, asset.Spot)
+	require.NoError(t, err, "IsPairAvailable must not error")
+	assert.False(t, ok, "IsPairAvailable should return false for non-existent pair")
+
+	_, err = pm.IsPairEnabled(cp, asset.PerpetualSwap)
+	assert.ErrorIs(t, err, ErrAssetNotFound, "Should error when asset is not found")
+
+	_, err = pm.IsPairEnabled(cp, asset.Item(1337))
+	assert.ErrorIs(t, err, asset.ErrNotSupported, "Should error when asset is not supported")
+
+	pm.Pairs[asset.PerpetualSwap] = &PairStore{}
+	_, err = pm.IsPairEnabled(cp, asset.PerpetualSwap)
+	assert.ErrorIs(t, err, ErrAssetIsNil, "Should error when store AssetEnabled is nil")
+
+	_, err = pm.IsPairEnabled(EMPTYPAIR, asset.PerpetualSwap)
+	assert.ErrorIs(t, err, ErrCurrencyPairEmpty, "Should error when currency pair is empty")
```

Comment:
I am going to assume a typo, or I'm not understanding why you're testing something else 😄 
```suggestion
	ok, err = pm.IsPairAvailable(NewPair(SAFE, MOONRISE), asset.Spot)
	require.NoError(t, err, "pm.IsPairAvailable must not error")
	assert.False(t, ok, "pm.IsPairAvailable should return correct value for an non-existent")

	ok, err = pm.IsPairAvailable(cp, asset.Futures)
	require.NoError(t, err, "pm.IsPairAvailable must not error")
	assert.False(t, ok, "pm.IsPairAvailable should return false for a disabled asset type")

	cp = NewPairWithDelimiter("XRP", "DOGE", "-")
	ok, err = pm.IsPairAvailable(cp, asset.Spot)
	require.NoError(t, err, "pm.IsPairAvailable must not error")
	assert.False(t, ok, "pm.IsPairAvailable should return false for non-existent pair")

	_, err = pm.IsPairAvailable(cp, asset.PerpetualSwap)
	assert.ErrorIs(t, err, ErrAssetNotFound, "Should error when asset is not found")

	_, err = pm.IsPairAvailable(cp, asset.Item(1337))
	assert.ErrorIs(t, err, asset.ErrNotSupported, "Should error when asset is not supported")

	pm.Pairs[asset.PerpetualSwap] = &PairStore{}
	_, err = pm.IsPairAvailable(cp, asset.PerpetualSwap)
	assert.ErrorIs(t, err, ErrAssetIsNil, "Should error when store AssetEnabled is nil")

	_, err = pm.IsPairAvailable(EMPTYPAIR, asset.PerpetualSwap)
	assert.ErrorIs(t, err, ErrCurrencyPairEmpty, "Should error when currency pair is empty")
```

Reply:
Damn. Yeah. Thanks.
I went to abstract it into closure over a method and test either of them, and then backed away from the crazy glue. In my haste to escape the intoxication of relentless abstraction I forgot to actually do it *right* 🤦 

Fixed ba314737